apps/pkeyutl.c

```diff
@@ -369,6 +373,7 @@ static void usage()
 #ifndef OPENSSL_NO_ENGINE
     BIO_printf(bio_err,
                "-engine e       use engine e, possibly a hardware device.\n");
+    BIO_printf(bio_err, "-engine_impl    access key through the engine\n");
```

Comment:
Unfortunately, those messages are misleading...  actually contrary to what they're doing.  May I suggest the following?

        BIO_printf(bio_err,
                   "-engine -e        use engine e, maybe a hardware device, for loading keys.\n");
        BIO_printf(bio_err, "-engine_impl    also use engine given by -engine for crypto operations\n");